bevy_lint/tests/ui.rs

```diff
@@ -55,3 +71,55 @@ fn config() -> color_eyre::Result<Config> {
 
     Ok(config)
 }
+
+#[derive(Deserialize, Debug)]
+#[serde(rename = "compiler-artifact", tag = "reason")]
+struct ArtifactMessage<'a> {
+    package_id: &'a str,
+    target: ArtifactTarget<'a>,
+    filenames: Vec<&'a Path>,
+}
+
+#[derive(Deserialize, Debug)]
+struct ArtifactTarget<'a> {
+    name: &'a str,
+    kind: Vec<&'a str>,
+}
+
+fn find_bevy_rlib() -> color_eyre::Result<PathBuf> {
+    let output = Command::new("cargo")
+        .arg("build")
+        .arg("--test=ui")
+        .arg("--message-format=json")
+        .stderr(Stdio::inherit())
+        .output()?;
+
+    ensure!(output.status.success());
+
+    const NEWLINE: u8 = '\n' as u8;
+    const BEVY_PACKAGE_ID_PREFIX: &str =
+        "registry+https://github.com/rust-lang/crates.io-index#bevy@";
+
+    let mut messages = Vec::with_capacity(1);
+
+    for line in output.stdout.split(|&byte| byte == NEWLINE) {
```

Comment:
Is it safe to split just by the bytes definition in Unicode? Or should we first parse into a string and use the `.lines()` function?

Reply:
That's a great point, I forgot that Unicode has a variable width! I'm going to swap this to use [`String::from_utf8_lossy()`](https://doc.rust-lang.org/stable/std/string/struct.String.html#method.from_utf8_lossy).